src/Slot-Core/Slot.class.st

```diff
@@ -256,7 +257,8 @@ Slot >> isVisible [
 
 { #category : #testing }
 Slot >> isWrittenIn: aCompiledCode [
-	^aCompiledCode writesSlot: self
+	^aCompiledCode ast instanceVariableWriteNodes
+		anySatisfy: [ :node | node binding slot == self ]
```

Comment:
Maybe we can introduce two methods for AST-nodes:
- aNode readsSlot: aSlot
- aNode writesSlot: aSlot
It will do recursion over children as default implementation on RBProgramNode. 
RBVariableNode will do "binding slot == aSlot"
RBAssignmentNode will provide special implementation for writesSlot: